stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/OTPElementUI.kt

```diff
@@ -196,7 +196,11 @@ private fun OTPInputBox(
             selection = if (isSelected) { TextRange(value.length) } else { TextRange.Zero }
         ),
         onValueChange = {
-            val inputLength = element.controller.onValueChanged(index, it.text)
+            // If the OTPInputBox already has a value, it would be the first character of it.text
+            // remove it before passing it to the controller.
+            val newValue =
```

Comment:
when OTP is already filled, inputting another char here would have `it.text` = `previousChar+newValue`, we'll need to remove the previousChar.

Note: newValue could be a single char _or_ pasted from clipboard